peps/pep-0791.rst

```diff
@@ -0,0 +1,158 @@
+return values are integers." could tell the truth once.  In a similar way we
+can simplify description of accepted arguments for both the
+:external+py3.14:mod:`math` and the new module.
+
+Apparently, the :external+py3.14:mod:`math` can't serve as a catch-all place
+for mathematical functions: we have also the :external+py3.14:mod:`cmath` and
+the :external+py3.14:mod:`statistics`.  Let's make same for integer-related
+functions.  It would provide shared context, which reduces verbosity in the
+documentation and conceptual load.  It also aids discoverability through
+grouping related functions and IDEs suggesting helpful completions.
+
+
+Specification
+=============
+
+The PEP proposes moving the following integer-related functions in a new
+module, called ``imath``:
+
+* :external+py3.14:func:`~math.comb`
+* :external+py3.14:func:`~math.factorial`
+* :external+py3.14:func:`~math.gcd`
+* :external+py3.14:func:`~math.isqrt`
+* :external+py3.14:func:`~math.lcm`
+* :external+py3.14:func:`~math.perm`
+
+Their aliases in :external+py3.14:mod:`math` will be :term:`soft deprecated`.
+
+Modules functions will accept integers and objects that implement the
+:external+py3.14:meth:`~object.__index__` method which is used to convert the
+object to an integer number.
+
+Possible extensions for the new module and its scope are discussed in the
+`Open Issues <Open Issues_>`_ section.  New functions are not part of the
+proposal.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for an indefinite time
+(their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+https://github.com/python/cpython/pull/133909
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+Chosen name seems consistent with other domain-specific mathematical module:
+:external+py3.14:mod:`cmath` (for complex numbers).
+
+There is already an ``imath`` project on PyPI, but only with two releases, with
+the most recent one four years ago.  Its repository is no longer accessible.
+The `Imath <https://github.com/AcademySoftwareFoundation/Imath>`_ C++ library
+include Python bindings with the same name.
+
+`Polling shows <https://discuss.python.org/t/91337/35>`_ ``intmath`` as another
+popular name.  The argument made was that the normal mathematical spelling of
+the imaginary unit is ``i``, which makes imath ambiguous.  It also has no conflict
+with any PyPI module.  On the other hand, ``intmath`` may be confused with
+interval math or numerical integration.
+
+Other proposed names include ``ntheory`` (like SymPy's submodule),
+``integermath`` and ``imaths``.
+
+
+Module scope and possible extensions
+------------------------------------
+
+Unless we can just provide bindings to some well supported mathematical library
+like the GMP, the module scope should be limited.  For example, no efficient primality
+testing and factorization.
```

Comment:
Steve D'Aprano [wrote](https://mail.python.org/archives/list/python-ideas@python.org/message/TAHFVIU7EQ5UYEQ52CCGSK4IYORZYT4U/):

"Primality testing and generation of primes is a bigger job than it might 
seem, probably deserving of an entire library. (Not necessarily in the 
std lib.)"

IMO it's a good summary. See the length of https://mail.python.org/archives/list/python-ideas@python.org/thread/YYJ5YJBJNCVXQWK5K3WSVNMPUSV56LOR/ discussion when it comes to prime numbers.